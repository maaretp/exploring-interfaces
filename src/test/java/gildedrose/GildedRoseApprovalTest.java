package gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.legacycode.Range;

import junit.framework.TestCase;

public class GildedRoseApprovalTest extends TestCase {

	public void testSomething() throws Exception {
		String[] items = { "something","Aged Brie", "Sulfuras", "Sulfuras, Hand of Ragnaros", "Backstage passes to a TAFKAL80ETC concert"};
		Integer[] sellins = {0, 1, 5, 6, 10, 50, Integer.MAX_VALUE};
		Integer[] qualities = {-1, 0, 50, 61, 80, Integer.MAX_VALUE};
		CombinationApprovals.verifyAllCombinations(this::checkItem, items, sellins, qualities);
	}
	
	public Item checkItem(String item, Integer sellin, Integer quality) {
		Item[] items = new Item[] { new Item(item, sellin, quality) };
		GildedRose app = new GildedRose(items);
		app.updateQuality();
		return items[0];
	}
	

	 //Once the sell by date has passed, Quality degrades twice as fast
		//- The Quality of an item is never negative
		//- "Aged Brie" actually increases in Quality the older it gets
		//- The Quality of an item is never more than 50
		//- "Sulfuras", being a legendary item, never has to be sold or decreases in Quality
		//- "Backstage passes", like aged brie, increases in Quality as its SellIn value approaches;
		//Quality increases by 2 when there are 10 days or less and by 3 when there are 5 days or less but
		//Quality drops to 0 after the concert
		//an item can never have its Quality increase above 50, 
		//however "Sulfuras" is a legendary item and as such its Quality is 80 and it never alters
	

}