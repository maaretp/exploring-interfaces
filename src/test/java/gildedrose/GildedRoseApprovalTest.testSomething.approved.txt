[something, 0, -1] => something, -1, -1 
[something, 0, 0] => something, -1, 0 
[something, 0, 50] => something, -1, 48 
[something, 0, 61] => something, -1, 59 
[something, 0, 80] => something, -1, 78 
[something, 0, 2147483647] => something, -1, 2147483645 
[something, 1, -1] => something, 0, -1 
[something, 1, 0] => something, 0, 0 
[something, 1, 50] => something, 0, 49 
[something, 1, 61] => something, 0, 60 
[something, 1, 80] => something, 0, 79 
[something, 1, 2147483647] => something, 0, 2147483646 
[something, 5, -1] => something, 4, -1 
[something, 5, 0] => something, 4, 0 
[something, 5, 50] => something, 4, 49 
[something, 5, 61] => something, 4, 60 
[something, 5, 80] => something, 4, 79 
[something, 5, 2147483647] => something, 4, 2147483646 
[something, 6, -1] => something, 5, -1 
[something, 6, 0] => something, 5, 0 
[something, 6, 50] => something, 5, 49 
[something, 6, 61] => something, 5, 60 
[something, 6, 80] => something, 5, 79 
[something, 6, 2147483647] => something, 5, 2147483646 
[something, 10, -1] => something, 9, -1 
[something, 10, 0] => something, 9, 0 
[something, 10, 50] => something, 9, 49 
[something, 10, 61] => something, 9, 60 
[something, 10, 80] => something, 9, 79 
[something, 10, 2147483647] => something, 9, 2147483646 
[something, 50, -1] => something, 49, -1 
[something, 50, 0] => something, 49, 0 
[something, 50, 50] => something, 49, 49 
[something, 50, 61] => something, 49, 60 
[something, 50, 80] => something, 49, 79 
[something, 50, 2147483647] => something, 49, 2147483646 
[something, 2147483647, -1] => something, 2147483646, -1 
[something, 2147483647, 0] => something, 2147483646, 0 
[something, 2147483647, 50] => something, 2147483646, 49 
[something, 2147483647, 61] => something, 2147483646, 60 
[something, 2147483647, 80] => something, 2147483646, 79 
[something, 2147483647, 2147483647] => something, 2147483646, 2147483646 
[Aged Brie, 0, -1] => Aged Brie, -1, 1 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 0, 61] => Aged Brie, -1, 61 
[Aged Brie, 0, 80] => Aged Brie, -1, 80 
[Aged Brie, 0, 2147483647] => Aged Brie, -1, 2147483647 
[Aged Brie, 1, -1] => Aged Brie, 0, 0 
[Aged Brie, 1, 0] => Aged Brie, 0, 1 
[Aged Brie, 1, 50] => Aged Brie, 0, 50 
[Aged Brie, 1, 61] => Aged Brie, 0, 61 
[Aged Brie, 1, 80] => Aged Brie, 0, 80 
[Aged Brie, 1, 2147483647] => Aged Brie, 0, 2147483647 
[Aged Brie, 5, -1] => Aged Brie, 4, 0 
[Aged Brie, 5, 0] => Aged Brie, 4, 1 
[Aged Brie, 5, 50] => Aged Brie, 4, 50 
[Aged Brie, 5, 61] => Aged Brie, 4, 61 
[Aged Brie, 5, 80] => Aged Brie, 4, 80 
[Aged Brie, 5, 2147483647] => Aged Brie, 4, 2147483647 
[Aged Brie, 6, -1] => Aged Brie, 5, 0 
[Aged Brie, 6, 0] => Aged Brie, 5, 1 
[Aged Brie, 6, 50] => Aged Brie, 5, 50 
[Aged Brie, 6, 61] => Aged Brie, 5, 61 
[Aged Brie, 6, 80] => Aged Brie, 5, 80 
[Aged Brie, 6, 2147483647] => Aged Brie, 5, 2147483647 
[Aged Brie, 10, -1] => Aged Brie, 9, 0 
[Aged Brie, 10, 0] => Aged Brie, 9, 1 
[Aged Brie, 10, 50] => Aged Brie, 9, 50 
[Aged Brie, 10, 61] => Aged Brie, 9, 61 
[Aged Brie, 10, 80] => Aged Brie, 9, 80 
[Aged Brie, 10, 2147483647] => Aged Brie, 9, 2147483647 
[Aged Brie, 50, -1] => Aged Brie, 49, 0 
[Aged Brie, 50, 0] => Aged Brie, 49, 1 
[Aged Brie, 50, 50] => Aged Brie, 49, 50 
[Aged Brie, 50, 61] => Aged Brie, 49, 61 
[Aged Brie, 50, 80] => Aged Brie, 49, 80 
[Aged Brie, 50, 2147483647] => Aged Brie, 49, 2147483647 
[Aged Brie, 2147483647, -1] => Aged Brie, 2147483646, 0 
[Aged Brie, 2147483647, 0] => Aged Brie, 2147483646, 1 
[Aged Brie, 2147483647, 50] => Aged Brie, 2147483646, 50 
[Aged Brie, 2147483647, 61] => Aged Brie, 2147483646, 61 
[Aged Brie, 2147483647, 80] => Aged Brie, 2147483646, 80 
[Aged Brie, 2147483647, 2147483647] => Aged Brie, 2147483646, 2147483647 
[Sulfuras, 0, -1] => Sulfuras, -1, -1 
[Sulfuras, 0, 0] => Sulfuras, -1, 0 
[Sulfuras, 0, 50] => Sulfuras, -1, 48 
[Sulfuras, 0, 61] => Sulfuras, -1, 59 
[Sulfuras, 0, 80] => Sulfuras, -1, 78 
[Sulfuras, 0, 2147483647] => Sulfuras, -1, 2147483645 
[Sulfuras, 1, -1] => Sulfuras, 0, -1 
[Sulfuras, 1, 0] => Sulfuras, 0, 0 
[Sulfuras, 1, 50] => Sulfuras, 0, 49 
[Sulfuras, 1, 61] => Sulfuras, 0, 60 
[Sulfuras, 1, 80] => Sulfuras, 0, 79 
[Sulfuras, 1, 2147483647] => Sulfuras, 0, 2147483646 
[Sulfuras, 5, -1] => Sulfuras, 4, -1 
[Sulfuras, 5, 0] => Sulfuras, 4, 0 
[Sulfuras, 5, 50] => Sulfuras, 4, 49 
[Sulfuras, 5, 61] => Sulfuras, 4, 60 
[Sulfuras, 5, 80] => Sulfuras, 4, 79 
[Sulfuras, 5, 2147483647] => Sulfuras, 4, 2147483646 
[Sulfuras, 6, -1] => Sulfuras, 5, -1 
[Sulfuras, 6, 0] => Sulfuras, 5, 0 
[Sulfuras, 6, 50] => Sulfuras, 5, 49 
[Sulfuras, 6, 61] => Sulfuras, 5, 60 
[Sulfuras, 6, 80] => Sulfuras, 5, 79 
[Sulfuras, 6, 2147483647] => Sulfuras, 5, 2147483646 
[Sulfuras, 10, -1] => Sulfuras, 9, -1 
[Sulfuras, 10, 0] => Sulfuras, 9, 0 
[Sulfuras, 10, 50] => Sulfuras, 9, 49 
[Sulfuras, 10, 61] => Sulfuras, 9, 60 
[Sulfuras, 10, 80] => Sulfuras, 9, 79 
[Sulfuras, 10, 2147483647] => Sulfuras, 9, 2147483646 
[Sulfuras, 50, -1] => Sulfuras, 49, -1 
[Sulfuras, 50, 0] => Sulfuras, 49, 0 
[Sulfuras, 50, 50] => Sulfuras, 49, 49 
[Sulfuras, 50, 61] => Sulfuras, 49, 60 
[Sulfuras, 50, 80] => Sulfuras, 49, 79 
[Sulfuras, 50, 2147483647] => Sulfuras, 49, 2147483646 
[Sulfuras, 2147483647, -1] => Sulfuras, 2147483646, -1 
[Sulfuras, 2147483647, 0] => Sulfuras, 2147483646, 0 
[Sulfuras, 2147483647, 50] => Sulfuras, 2147483646, 49 
[Sulfuras, 2147483647, 61] => Sulfuras, 2147483646, 60 
[Sulfuras, 2147483647, 80] => Sulfuras, 2147483646, 79 
[Sulfuras, 2147483647, 2147483647] => Sulfuras, 2147483646, 2147483646 
[Sulfuras, Hand of Ragnaros, 0, -1] => Sulfuras, Hand of Ragnaros, 0, -1 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 0, 61] => Sulfuras, Hand of Ragnaros, 0, 61 
[Sulfuras, Hand of Ragnaros, 0, 80] => Sulfuras, Hand of Ragnaros, 0, 80 
[Sulfuras, Hand of Ragnaros, 0, 2147483647] => Sulfuras, Hand of Ragnaros, 0, 2147483647 
[Sulfuras, Hand of Ragnaros, 1, -1] => Sulfuras, Hand of Ragnaros, 1, -1 
[Sulfuras, Hand of Ragnaros, 1, 0] => Sulfuras, Hand of Ragnaros, 1, 0 
[Sulfuras, Hand of Ragnaros, 1, 50] => Sulfuras, Hand of Ragnaros, 1, 50 
[Sulfuras, Hand of Ragnaros, 1, 61] => Sulfuras, Hand of Ragnaros, 1, 61 
[Sulfuras, Hand of Ragnaros, 1, 80] => Sulfuras, Hand of Ragnaros, 1, 80 
[Sulfuras, Hand of Ragnaros, 1, 2147483647] => Sulfuras, Hand of Ragnaros, 1, 2147483647 
[Sulfuras, Hand of Ragnaros, 5, -1] => Sulfuras, Hand of Ragnaros, 5, -1 
[Sulfuras, Hand of Ragnaros, 5, 0] => Sulfuras, Hand of Ragnaros, 5, 0 
[Sulfuras, Hand of Ragnaros, 5, 50] => Sulfuras, Hand of Ragnaros, 5, 50 
[Sulfuras, Hand of Ragnaros, 5, 61] => Sulfuras, Hand of Ragnaros, 5, 61 
[Sulfuras, Hand of Ragnaros, 5, 80] => Sulfuras, Hand of Ragnaros, 5, 80 
[Sulfuras, Hand of Ragnaros, 5, 2147483647] => Sulfuras, Hand of Ragnaros, 5, 2147483647 
[Sulfuras, Hand of Ragnaros, 6, -1] => Sulfuras, Hand of Ragnaros, 6, -1 
[Sulfuras, Hand of Ragnaros, 6, 0] => Sulfuras, Hand of Ragnaros, 6, 0 
[Sulfuras, Hand of Ragnaros, 6, 50] => Sulfuras, Hand of Ragnaros, 6, 50 
[Sulfuras, Hand of Ragnaros, 6, 61] => Sulfuras, Hand of Ragnaros, 6, 61 
[Sulfuras, Hand of Ragnaros, 6, 80] => Sulfuras, Hand of Ragnaros, 6, 80 
[Sulfuras, Hand of Ragnaros, 6, 2147483647] => Sulfuras, Hand of Ragnaros, 6, 2147483647 
[Sulfuras, Hand of Ragnaros, 10, -1] => Sulfuras, Hand of Ragnaros, 10, -1 
[Sulfuras, Hand of Ragnaros, 10, 0] => Sulfuras, Hand of Ragnaros, 10, 0 
[Sulfuras, Hand of Ragnaros, 10, 50] => Sulfuras, Hand of Ragnaros, 10, 50 
[Sulfuras, Hand of Ragnaros, 10, 61] => Sulfuras, Hand of Ragnaros, 10, 61 
[Sulfuras, Hand of Ragnaros, 10, 80] => Sulfuras, Hand of Ragnaros, 10, 80 
[Sulfuras, Hand of Ragnaros, 10, 2147483647] => Sulfuras, Hand of Ragnaros, 10, 2147483647 
[Sulfuras, Hand of Ragnaros, 50, -1] => Sulfuras, Hand of Ragnaros, 50, -1 
[Sulfuras, Hand of Ragnaros, 50, 0] => Sulfuras, Hand of Ragnaros, 50, 0 
[Sulfuras, Hand of Ragnaros, 50, 50] => Sulfuras, Hand of Ragnaros, 50, 50 
[Sulfuras, Hand of Ragnaros, 50, 61] => Sulfuras, Hand of Ragnaros, 50, 61 
[Sulfuras, Hand of Ragnaros, 50, 80] => Sulfuras, Hand of Ragnaros, 50, 80 
[Sulfuras, Hand of Ragnaros, 50, 2147483647] => Sulfuras, Hand of Ragnaros, 50, 2147483647 
[Sulfuras, Hand of Ragnaros, 2147483647, -1] => Sulfuras, Hand of Ragnaros, 2147483647, -1 
[Sulfuras, Hand of Ragnaros, 2147483647, 0] => Sulfuras, Hand of Ragnaros, 2147483647, 0 
[Sulfuras, Hand of Ragnaros, 2147483647, 50] => Sulfuras, Hand of Ragnaros, 2147483647, 50 
[Sulfuras, Hand of Ragnaros, 2147483647, 61] => Sulfuras, Hand of Ragnaros, 2147483647, 61 
[Sulfuras, Hand of Ragnaros, 2147483647, 80] => Sulfuras, Hand of Ragnaros, 2147483647, 80 
[Sulfuras, Hand of Ragnaros, 2147483647, 2147483647] => Sulfuras, Hand of Ragnaros, 2147483647, 2147483647 
[Backstage passes to a TAFKAL80ETC concert, 0, -1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 61] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 80] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 2147483647] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 1, -1] => Backstage passes to a TAFKAL80ETC concert, 0, 2 
[Backstage passes to a TAFKAL80ETC concert, 1, 0] => Backstage passes to a TAFKAL80ETC concert, 0, 3 
[Backstage passes to a TAFKAL80ETC concert, 1, 50] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 1, 61] => Backstage passes to a TAFKAL80ETC concert, 0, 61 
[Backstage passes to a TAFKAL80ETC concert, 1, 80] => Backstage passes to a TAFKAL80ETC concert, 0, 80 
[Backstage passes to a TAFKAL80ETC concert, 1, 2147483647] => Backstage passes to a TAFKAL80ETC concert, 0, 2147483647 
[Backstage passes to a TAFKAL80ETC concert, 5, -1] => Backstage passes to a TAFKAL80ETC concert, 4, 2 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 61] => Backstage passes to a TAFKAL80ETC concert, 4, 61 
[Backstage passes to a TAFKAL80ETC concert, 5, 80] => Backstage passes to a TAFKAL80ETC concert, 4, 80 
[Backstage passes to a TAFKAL80ETC concert, 5, 2147483647] => Backstage passes to a TAFKAL80ETC concert, 4, 2147483647 
[Backstage passes to a TAFKAL80ETC concert, 6, -1] => Backstage passes to a TAFKAL80ETC concert, 5, 1 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => Backstage passes to a TAFKAL80ETC concert, 5, 2 
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 61] => Backstage passes to a TAFKAL80ETC concert, 5, 61 
[Backstage passes to a TAFKAL80ETC concert, 6, 80] => Backstage passes to a TAFKAL80ETC concert, 5, 80 
[Backstage passes to a TAFKAL80ETC concert, 6, 2147483647] => Backstage passes to a TAFKAL80ETC concert, 5, 2147483647 
[Backstage passes to a TAFKAL80ETC concert, 10, -1] => Backstage passes to a TAFKAL80ETC concert, 9, 1 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 61] => Backstage passes to a TAFKAL80ETC concert, 9, 61 
[Backstage passes to a TAFKAL80ETC concert, 10, 80] => Backstage passes to a TAFKAL80ETC concert, 9, 80 
[Backstage passes to a TAFKAL80ETC concert, 10, 2147483647] => Backstage passes to a TAFKAL80ETC concert, 9, 2147483647 
[Backstage passes to a TAFKAL80ETC concert, 50, -1] => Backstage passes to a TAFKAL80ETC concert, 49, 0 
[Backstage passes to a TAFKAL80ETC concert, 50, 0] => Backstage passes to a TAFKAL80ETC concert, 49, 1 
[Backstage passes to a TAFKAL80ETC concert, 50, 50] => Backstage passes to a TAFKAL80ETC concert, 49, 50 
[Backstage passes to a TAFKAL80ETC concert, 50, 61] => Backstage passes to a TAFKAL80ETC concert, 49, 61 
[Backstage passes to a TAFKAL80ETC concert, 50, 80] => Backstage passes to a TAFKAL80ETC concert, 49, 80 
[Backstage passes to a TAFKAL80ETC concert, 50, 2147483647] => Backstage passes to a TAFKAL80ETC concert, 49, 2147483647 
[Backstage passes to a TAFKAL80ETC concert, 2147483647, -1] => Backstage passes to a TAFKAL80ETC concert, 2147483646, 0 
[Backstage passes to a TAFKAL80ETC concert, 2147483647, 0] => Backstage passes to a TAFKAL80ETC concert, 2147483646, 1 
[Backstage passes to a TAFKAL80ETC concert, 2147483647, 50] => Backstage passes to a TAFKAL80ETC concert, 2147483646, 50 
[Backstage passes to a TAFKAL80ETC concert, 2147483647, 61] => Backstage passes to a TAFKAL80ETC concert, 2147483646, 61 
[Backstage passes to a TAFKAL80ETC concert, 2147483647, 80] => Backstage passes to a TAFKAL80ETC concert, 2147483646, 80 
[Backstage passes to a TAFKAL80ETC concert, 2147483647, 2147483647] => Backstage passes to a TAFKAL80ETC concert, 2147483646, 2147483647 
